src/main/java/ch/njol/skript/expressions/ExprNamed.java

```diff
@@ -35,60 +35,84 @@
+	"open hopper inventory named \"Magic Hopper\" to player"
+})
 @Since("2.0, 2.2-dev34 (inventories)")
 public class ExprNamed extends PropertyExpression<Object, Object> {
+
 	static {
-		Skript.registerExpression(ExprNamed.class, Object.class, ExpressionType.PROPERTY,
+		Skript.registerExpression(ExprNamed.class, Object.class, ExpressionType.COMBINED,
 				"%itemtype/inventorytype% (named|with name[s]) %string%");
 	}
-	
-	@SuppressWarnings("null")
+
 	private Expression<String> name;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		setExpr(exprs[0]);
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		// See issue #1822 as to why this convert is required.
+		if (exprs[0].getReturnType().equals(ItemStack.class)) {
+			setExpr(exprs[0].getConvertedExpression(ItemType.class));
+		} else {
+			setExpr(exprs[0]);
+		}
 		name = (Expression<String>) exprs[1];
+		if (getExpr() instanceof Literal) {
+			Literal<?> literal = (Literal<?>) getExpr();
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !((InventoryType) object).isCreatable()) {
+				// Spigot forgot to label some InventoryType's as non creatable in some versions < 1.19.4
+				// So this throws NullPointerException aswell ontop of the IllegalArgumentException.
+				// See https://hub.spigotmc.org/jira/browse/SPIGOT-7301
+				Skript.error("You can't create a '" + literal.toString() + "' inventory. It's not creatable!");
+				return false;
+			}
+		}
 		return true;
 	}
-	
+
 	@Override
-	protected Object[] get(final Event e, final Object[] source) {
-		String name = this.name.getSingle(e);
+	protected Object[] get(Event event, Object[] source) {
+		String name = this.name.getSingle(event);
 		if (name == null)
-			return get(source, obj -> obj); // No name provided, do nothing
+			return get(source, object -> {
+				if (object instanceof InventoryType) {
+					try {
+						return Bukkit.createInventory(null, (InventoryType) object);
+					} catch (NullPointerException | IllegalArgumentException e) {
+						return null;
+					}
+				}
+				return object; // Return the same ItemType they passed without applying a name.
+			});
 		return get(source, new Getter<Object, Object>() {
 			@Override
 			@Nullable
-			public Object get(Object obj) {
-				if (obj instanceof InventoryType)
-					return Bukkit.createInventory(null, (InventoryType) obj, name);
-				if (obj instanceof ItemStack) {
-					ItemStack stack = (ItemStack) obj;
-					stack = stack.clone();
-					ItemMeta meta = stack.getItemMeta();
-					if (meta != null) {
-						meta.setDisplayName(name);
-						stack.setItemMeta(meta);
+			public Object get(Object object) {
+				if (object instanceof InventoryType) {
+					try {
```

Comment:
Same here, I think we'd be best off at least checking if it is creatable first. Since we have to do this in two places, and have the try (maybe could be replaced with checking incorrectly marked types? not sure), it might be worth moving to a private method